bam_fastq.c

```diff
@@ -260,9 +261,99 @@ static int getLength(char **s)
     return n;
 }
 
+/*
+ * Expand up a B aux tag to string form.
+ * Modelled on sam_format1_append.
+ *
+ * Returns 0 on success,
+ *        -1 on failure
+ */
+static int print_aux_B(uint8_t *s, uint8_t *s_end, kstring_t *str) {
+    int r = 0, i;
+
+    if (s_end - s < 6) return -1;
+
+    int sub_type = *++s;
+    uint32_t n;
+    n = le_to_u32(++s);
+    s += 4;
+
+    r |= kputc_(sub_type, str);
+    switch (sub_type) {
+    case 'c':
+        if (s_end - s < n) return -1;
+        if (ks_resize(str, str->l + n*2) < 0) return -1;
```

Comment:
This is just copied directly out of sam.c.  I'd rather not obfuscate the source by having two different ways of doing the same thing.

Ideally we'd have a common function in htslib; an API that stringifies any tag so it can be called by this code, by htslib SAM formatter, and anything else, but that's for a subsequent piece of work I think.

Reply:
Actually, it's pointless making changes in two places and it's clear the original code has flaws too given the complaints raised here.

I'll look at a common API shared by both.